Lesson16/Mesut/arraysAndObjectsManipulations.js

```diff
@@ -4,72 +4,164 @@
 // Input: [1, 2, 3, 4]
 // Output: 10
 
-console.log('----------------------------');
+console.log('-----------exercise 2-----------------');
 // 2. Write a function that takes an array of integers and returns the average of all elements.
 //Input: [1, 2, 3, 4]
 // Output: 2.5
 
-console.log('----------------------------');
+console.log('------------exercise 3----------------');
 // 3. Write a function that takes an array of strings and concatenates them together adding spaces.
 // Input: ["Hello", "World", "!"]
 // Output: "Hello World!"
-
-console.log('----------------------------');
+function arrToStr(arr){
+    let string="";
+    
+    for (let i = 0; i < arr.length; i++) {
+    string=string+" "+arr[i];
+        
+    }
+console.log(string);
+}
+arrToStr(["Hello", "World", "!"])
+console.log('-----------exercise 4-----------------');
 // 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
 // Input: [1, 2, 3, 4, 7, 10];
 // Output: [1, 3, 7]
-
-console.log('----------------------------');
+function oddNumbers(arr){
+    let newArr=[]
+for(let num in arr){
+    if(arr[num]%2==!0){
+        newArr.push(arr[num]);
+    }
+   
+}
+console.log(newArr);
+}
+oddNumbers([1, 2, 3, 4, 7, 10])
+console.log('------------exercise 5----------------');
 // 5. Write a function that takes an array of integers and returns a new array with even numbers only.
 // Input: [1, 2, 3, 4, 7, 10]
 // Output: [1, 2, 4, 10]
 
-console.log('----------------------------');
+function evenNumbers(arr){
+    let newArr2=[];
```

Comment:
I suppose you named this variable `newArr2` as you had a variable named `newArr` somewhere else.

Since this variable (`newArr2`)  is inside `evenNumbers` function (also known as in "local scope"), it can be `newArr` without resulting in an error.

No need to change, just want to make sure that you understand the difference between global and local scope. :)